common/build/eslint-plugin-fluid/src/test/example/no-unchecked-record-access/indexedRecordOfStrings.ts

```diff
@@ -112,47 +76,13 @@ for (const key of Object.keys(indexedRecordOfStrings)) {
 const aExpectingString: string = indexedRecordOfStrings.a; // defect: Assigning index property 'a' to a strict string variable, but 'a' might not be present
 const aExpectingStringOrUndefined: string | undefined = indexedRecordOfStrings.a; // ok: Assigning index property 'a' to string or undefined variable, 'a' might not be present
 const aImplicitType = indexedRecordOfStrings.a; // defect: Assigning index property with inferred type
-aImplicitType.length; // ok: This should be caught by tsc, not by this custom lint rule
-aImplicitType?.length; // ok: This should be caught by tsc, not by this custom lint rule
-aImplicitType!.length; // ok: This should be caught by tsc, not by this custom lint rule
+aImplicitType.length; // ok: This should be caught by the instantiation in the variable declaration
+aImplicitType?.length; // ok: This should be caught by the instantiation in the variable declaration
+aImplicitType!.length; // ok: This should be caught by the instantiation in the variable declaration
```

Comment:
I would drop lines 80 and 81. They aren't doing anything.
For 79, I would update the comment to say it is continuation of the inferred type case. (There is nothing to be caught on that line - certainly the "this should be caught" doesn't mean anything - there is no "this".)